apps/jupyter/extensions/notebook-runner/src/iframe-api.ts

```diff
@@ -508,50 +504,49 @@ export class EmbeddedPythonCallbacks {
     }
   }
 
-  private async writeFolderContents(
-    basePath: string,
-    files: Directory,
-  ): Promise<void> {
-    for (const [relativePath, blob] of files.entries()) {
-      const path = `${basePath}/${relativePath}`;
-      await this.putFileContents(path, blob);
-    }
-  }
-
-  private async readFolderContents(basePath: string): Promise<Directory> {
+  private async getFolderContents(path: string): Promise<Map<string, Blob>> {
     const files = new Map<string, Blob>();
 
     try {
-      const folder = await this.app.serviceManager.contents.get(basePath, {
+      const folder = await this.app.serviceManager.contents.get(path, {
         content: true,
       });
 
       if (folder.type !== "directory") {
-        throw new DirectoryNotFoundError(basePath);
+        throw new DirectoryNotFoundError(path);
       }
 
-      for (const item of folder.content || []) {
-        const itemPath = `${basePath}/${item.name}`;
+      for (const item of folder.content) {
+        const itemPath = `${path}/${item.name}`;
 
         if (item.type === "directory") {
-          const subFiles = await this.readFolderContents(itemPath);
-
+          // Recursively read subdirectories
+          const subFiles = await this.getFolderContents(itemPath);
           for (const [subPath, blob] of subFiles.entries()) {
             files.set(`${item.name}/${subPath}`, blob);
           }
-        } else {
-          const blob = await this.getFileContents(itemPath);
-
-          files.set(item.name, blob);
+        } else if (item.type === "file") {
```

Comment:
What if it is not a directory or a file? Do we want to throw or at least log something? And is there right now a different type?